### PR TITLE
Redone weapon dropping (with concommands)

### PR DESCRIPTION
**Pull request by overlordbleck**
Revamped (hopefully better)

### CODE REVIEW

**Comment by overlordbleck** [comment]
Untested, please wait before merging!

File: lua/autorun/client/cl_pvp_movespeed.lua [hunk 1]
```diff
@@ -1,2 +1,8 @@
 -- Includes --
-include( "autorun/shared/sh_pvp_movespeed.lua" )
\ No newline at end of file
+include( "autorun/shared/sh_pvp_movespeed.lua" )
+
+--Sends to server to drop current weapon
+concommand.Add( "cfc_dropweapon", function()
```
Comment (plally) [inline lua/autorun/client/cl_pvp_movespeed.lua:5]: console commands can be created serverside, so you don't need net message code.

Comment (overlordbleck) [reply lua/autorun/client/cl_pvp_movespeed.lua:5]: Created in the client so it shows up when you're typing it in the console. If it was created server side it wouldn't show the command when typing it.

File: lua/autorun/server/sv_pvp_movespeed.lua [hunk 2]
```diff
@@ -83,6 +96,26 @@ local function getPlayerWeight( ply )
...
 end
 
+local function dropPlyWeapon( ply )
+    local currentWeapon = ply:GetActiveWeapon():GetClass()
+    if isUndroppable[currentWeapon] then
+        ply:ChatPrint("This weapon is unable to be dropped!")
+        return
+    end
+
+    ply:StripWeapon( currentWeapon )
```
Comment (plally) [inline lua/autorun/server/sv_pvp_movespeed.lua]: couldn't you use player:DropWeapon here?

Comment (overlordbleck) [reply lua/autorun/server/sv_pvp_movespeed.lua]: Drop weapon doesn't return an entity so we can't despawn the weapon after a given time frame.

Comment (plally) [reply lua/autorun/server/sv_pvp_movespeed.lua]: the weapon entity doesn't change when its dropped

File: lua/autorun/server/sv_pvp_movespeed.lua [hunk 3]
```diff
@@ -98,9 +128,29 @@ local function onDrop( ply, wep )
     setSpeedFromWeight( ply, totalWeight )
 end
 
+local function onPlayerSay( ply, text )
+    if not IsValid( ply ) then return end
+    if not ply:Alive() then return end
+    
+    if isDropCommand[text] then 
+        dropPlyWeapon( ply )
+        return
```
Comment (sarahsturgeon) [inline lua/autorun/server/sv_pvp_movespeed.lua]:
```suggestion
```

Not necessary here

**Review by sarahsturgeon — DISMISSED** [review]
Quick change then it looks good!

**Comment by overlordbleck** [comment]
Ready for merge.

File: lua/autorun/server/sv_pvp_movespeed.lua [hunk 4]
```diff
@@ -83,6 +96,23 @@ local function getPlayerWeight( ply )
...
+    if isUndroppable[currentWeapon:GetClass()] then
+        ply:ChatPrint("This weapon is unable to be dropped!")
+        return
+    end
+
+    ply:DropWeapon( currentWeapon )
+    
+    currentWeapon.despawn = timer.Simple( 10, function()
+        if not IsValid( currentWeapon ) then return end
+        if IsValid( currentWeapon.Owner ) then return end
```
Comment (justinmuskopf) [inline lua/autorun/server/sv_pvp_movespeed.lua:110]: Do weapons not retain their owner when dropped?

File: lua/autorun/server/sv_pvp_movespeed.lua [hunk 5]
```diff
@@ -98,9 +128,29 @@ local function onDrop( ply, wep )
     setSpeedFromWeight( ply, totalWeight )
 end
 
+local function onPlayerSay( ply, text )
+    if not IsValid( ply ) then return end
+    if not ply:Alive() then return end
+    
+    if isDropCommand[text] then 
+        dropPlyWeapon( ply )
+        return ""
```
Comment (justinmuskopf) [inline lua/autorun/server/sv_pvp_movespeed.lua:137]: Is this how we keep it from going to chat?

Comment (overlordbleck) [reply lua/autorun/server/sv_pvp_movespeed.lua:137]: Yes, http://wiki.garrysmod.com/page/GM/PlayerSay it says so in the return section.

File: lua/autorun/server/sv_pvp_movespeed.lua (hunk 4 above)
Comment (overlordbleck) [reply lua/autorun/server/sv_pvp_movespeed.lua:110]: Its just so the weapon isn't removed when its being held by a player, like if they pick the weapon back up.

**Review by sarahsturgeon — APPROVED** [review]

File: lua/autorun/server/sv_pvp_movespeed.lua (hunk 5 above)
Comment (justinmuskopf) [reply lua/autorun/server/sv_pvp_movespeed.lua:137]: Neat!

**Review by justinmuskopf — APPROVED** [review]

**Pull request merged**